Source/WebCore/css/parser/CSSPropertyParser.cpp

```diff
@@ -826,9 +827,76 @@ static RefPtr<CSSPrimitiveValue> consumeFontVariantCaps(CSSParserTokenRange& ran
         CSSValueUnicase, CSSValueTitlingCaps>(range);
 }
 
-static RefPtr<CSSPrimitiveValue> consumeFontVariantAlternates(CSSParserTokenRange& range)
+static RefPtr<CSSValue> consumeFontVariantAlternates(CSSParserTokenRange& range)
 {
-    return consumeIdent<CSSValueNormal, CSSValueHistoricalForms>(range);
+    if (range.atEnd())
+        return nullptr;
+
+    if (range.peek().id() == CSSValueNormal) {
+        consumeIdent<CSSValueNormal>(range);
+        return CSSValuePool::singleton().createIdentifierValue(CSSValueNormal);
+    }
+
+    auto result = FontVariantAlternates::Normal();
+
+    auto parseSomethingWithoutError = [&range, &result]() {
+        bool hasParsedSomething = false;
+        auto parseAndSetArgument = [&range, &hasParsedSomething] (auto& value) {
+            CSSParserTokenRange args = consumeFunction(range);
+            auto ident = consumeCustomIdent(args);
+            if (!args.atEnd())
+                return false;
+        
+            if (!ident)
+                return false;
+        
+            if (value)
+                return false;
+        
+            hasParsedSomething = true;
+            value = ident->stringValue();
+            return true;
+        };
+        for (;;) {
```

Comment:
```suggestion
        while (true) {
```

would be better for consistency.

Reply:
Fixed